### PR TITLE
fixing routing config recovery after differential update errors

### PR DESCRIPTION
**Pull request by aryszka**
Fixing a bug where on route config updates, the recovery request for the full route config from a data client is not handled correctly, causing that routes that were deleted during the failing period are not removed from the active configuration.

### CODE REVIEW

File: routing/datasource_test.go [hunk 1]
```diff
@@ -101,7 +101,7 @@ func TestLogging(t *testing.T) {
 
 		testLog := loggingtest.New()
 		defer testLog.Close()
-		testLog.Unmute()
+		// testLog.Unmute()
```
Comment (mikkeloscar) [inline routing/datasource_test.go]: leftover debug code? :)

**Comment by szuecs** [comment]
👍

File: routing/datasource_test.go (hunk 1 above)
Comment (aryszka) [reply routing/datasource_test.go]: right!

**Comment by aryszka** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

File: routing/routing_test.go [hunk 2]
```diff
@@ -734,3 +735,70 @@ func TestRoutingHandlerHEAD(t *testing.T) {
 		t.Error("invalid count header")
 	}
 }
+
+func TestUpdateFailsRecovers(t *testing.T) {
+	l := loggingtest.New()
+	// l.Unmute()
```
Comment (mikkeloscar) [inline routing/routing_test.go]: nit: @aryszka also this one :X

Comment (aryszka) [reply routing/routing_test.go]: thanks! :smile:

**Comment by aryszka** [comment]
:+1:

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**